Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -593,15 +593,97 @@ Corellium allows you to launch multiple instances of a device (jailbroken or not
 
 An introduction to binary analysis using binary analysis frameworks has already been discussed in the "[Dynamic Analysis](0x05c-Reverse-Engineering-and-Tampering.md#dynamic-analysis "Dynamic analysis")" section for Android. We recommend you to revisit this section and refresh the concepts on this subject.
 
-For Android, we used Angr's symbolic execution engine to solve a challenge. In this section, we will revisit the Angr binary analysis framework to analyze the [UnCrackable Level 1 crackme app](https://github.com/OWASP/owasp-mstg/blob/master/Crackmes/iOS/Level_01/UnCrackable_Level1.ipa "UnCrackable Level 1 iOS App") but instead of symbolic execution we will use its concrete execution (or dynamic execution) features.
+For Android, we used Angr's symbolic execution engine to solve a challenge. In this section, we will firstly use Unicorn to solve [UnCrackable Level 1 crackme app](https://github.com/OWASP/owasp-mstg/blob/master/Crackmes/iOS/Level_01/UnCrackable_Level1.ipa "UnCrackable Level 1 iOS App") challenge and then we will revisit the Angr binary analysis framework to analyze the challenge but instead of symbolic execution we will use its concrete execution (or dynamic execution) features.
+
+### Unicorn
+
+Unicorn is a lightweight, multi-architecture CPU emulator framework. Unicorn provides us the basic infrastructure needed to execute processor instructions. In this section we will use Unicorn to solve UnCrackable Level 1 iOS App.
+
+To use Unicorn, we need to implement all necessary infrastructure which generally is readily available to us in an Operating System, which includes binary loader, linker and other dependency libraries. Writing a complete binary loader is superfluous for our current goal,  instead we will write our own crude binary loader and execute a small part of the binary in Unicorn to solve the challenge.
+
+While performing manual analysis in "[Reviewing Disassembled Native Code](#reviewing-disassembled-native-code "Reviewing Disassembled Native Code")" section, we determined that the function at offset 0x1000080d4 is responsible for dynamically generating the secret string. If we analyze this function and the subsequent function calls, we will observe that there is no hard dependency on any external library and neither its performing any system calls. All the necessary code is self-contained in the binary and this is a perfect scenario to use a CPU emulator like Unicorn.
```

Comment:
Are you referring to address 0x1000080d4? This is the function in 'manual_reversing_ghidra_native_disassembly.png'

Reply:
aah okay, u might be referring to 0x100008198. I will add a small snippet for that.

Reply:
I thought that maybe you could use another address from the current snippet picture as example instead of putting a new one. To keep it simple. Would that be possible?

Reply:
Argh, i should have seen this before. Done now. :)